src/main/java/io/weaviate/client/v1/schema/api/ClassExists.java

```diff
@@ -0,0 +1,50 @@
+package io.weaviate.client.v1.schema.api;
+
+import io.weaviate.client.Config;
+import io.weaviate.client.base.ClientResult;
+import io.weaviate.client.base.Result;
+import io.weaviate.client.base.WeaviateError;
+import io.weaviate.client.base.WeaviateErrorMessage;
+import io.weaviate.client.base.WeaviateErrorResponse;
+import io.weaviate.client.base.http.HttpClient;
+import io.weaviate.client.v1.schema.model.WeaviateClass;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpStatus;
+
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class ClassExists implements ClientResult<Boolean> {
+  private final ClassGetter classGetter;
+  private String className;
+
+  public ClassExists(HttpClient httpClient, Config config) {
+    this.classGetter = new ClassGetter(httpClient, config);
```

Comment:
Avoid creating instances of dependencies inside classes. Instead create dependencies outside and inject them as arguments of constructor.

It will make testing a lot easier by using mocks instead of real implementations.

Reply:
`this.` prefix is not necessary here